host/src/server/api/v3/proof/batch.rs

```diff
@@ -81,29 +82,43 @@ async fn batch_handler(
             batch_request.prover_args.clone().into(),
         )
         .into();
+
         sub_request_keys.push(request_key);
         sub_request_entities.push(request_entity);
         sub_batch_ids.push(*batch_id);
     }
 
-    let agg_request_key =
-        AggregationRequestKey::new(batch_request.proof_type, sub_batch_ids.clone()).into();
-    let agg_request_entity_without_proofs = AggregationRequestEntity::new(
-        sub_batch_ids,
-        vec![],
-        batch_request.proof_type,
-        batch_request.prover_args,
-    )
-    .into();
-    let result = prove_aggregation(
-        &actor,
-        agg_request_key,
-        agg_request_entity_without_proofs,
-        sub_request_keys,
-        sub_request_entities,
-    )
-    .await;
-
+    let result = if batch_request.aggregate.unwrap_or(false) {
+        prove_aggregation(
+            &actor,
+            AggregationRequestKey::new(batch_request.proof_type, sub_batch_ids.clone()).into(),
+            AggregationRequestEntity::new(
+                sub_batch_ids,
+                vec![],
+                batch_request.proof_type,
+                batch_request.prover_args,
+            )
+            .into(),
+            sub_request_keys,
+            sub_request_entities,
+        )
+        .await
+    } else {
+        prove_many(&actor, sub_request_keys, sub_request_entities)
```

Comment:
diff between prove_many & prove_aggregation is prove_many just save separate proof of each blocks, but prove_aggregation gives out aggregated prove, is it?

Reply:
Yes

Reply:
OK, let's merge after your tests pass.